File: android/app/src/main/java/com/sundeep/keypoints/NewsArticleNativeAdFactory.java
package com.sundeep.keypoints;

import android.view.LayoutInflater;
import android.view.View;
import android.widget.Button;
import android.widget.ImageView;
import android.widget.TextView;
import com.google.android.gms.ads.nativead.NativeAd;
import com.google.android.gms.ads.nativead.NativeAdView;
import io.flutter.plugins.googlemobileads.GoogleMobileAdsPlugin.NativeAdFactory;
import java.util.Map;

public class NewsArticleNativeAdFactory implements NativeAdFactory {
    private final LayoutInflater layoutInflater;

    public NewsArticleNativeAdFactory(LayoutInflater layoutInflater) {
        this.layoutInflater = layoutInflater;
    }

    @Override
    public NativeAdView createNativeAd(NativeAd nativeAd, Map<String, Object> customOptions) {
        // Inflate the native ad layout
        NativeAdView adView = (NativeAdView) layoutInflater.inflate(R.layout.native_ad_news_article, null);

        // Set up the ad view components
        TextView headlineView = adView.findViewById(R.id.ad_headline);
        TextView bodyView = adView.findViewById(R.id.ad_body);
        TextView advertiserView = adView.findViewById(R.id.ad_advertiser);
        Button callToActionView = adView.findViewById(R.id.ad_call_to_action);
        ImageView iconView = adView.findViewById(R.id.ad_icon);
        com.google.android.gms.ads.nativead.MediaView mediaView = adView.findViewById(R.id.ad_media);

        // Populate the ad view with native ad content
        if (nativeAd.getHeadline() != null) {
            headlineView.setText(nativeAd.getHeadline());
        }
        
        if (nativeAd.getBody() != null) {
            bodyView.setText(nativeAd.getBody());
        }
        
        if (nativeAd.getAdvertiser() != null) {
            advertiserView.setText(nativeAd.getAdvertiser());
        }
        
        if (nativeAd.getCallToAction() != null) {
            callToActionView.setText(nativeAd.getCallToAction());
        }
        
        if (nativeAd.getIcon() != null) {
            iconView.setImageDrawable(nativeAd.getIcon().getDrawable());
        }

        // Register the view components with the native ad view
        adView.setHeadlineView(headlineView);
        adView.setBodyView(bodyView);
        adView.setAdvertiserView(advertiserView);
        adView.setCallToActionView(callToActionView);
        adView.setIconView(iconView);
        adView.setMediaView(mediaView);

        // Set the native ad
        adView.setNativeAd(nativeAd);

        return adView;
    }
}

File: android/app/src/main/kotlin/com/example/keypoints/NewsArticleNativeAdFactory.kt
package com.example.keypoints

import android.view.LayoutInflater
import android.view.View
import android.widget.Button
import android.widget.ImageView
import android.widget.TextView
import com.google.android.gms.ads.nativead.NativeAd
import com.google.android.gms.ads.nativead.NativeAdView
import io.flutter.plugins.googlemobileads.GoogleMobileAdsPlugin.NativeAdFactory

class NewsArticleNativeAdFactory(private val layoutInflater: LayoutInflater) : NativeAdFactory {
    
    override fun createNativeAd(nativeAd: NativeAd, customOptions: MutableMap<String, Any>?): NativeAdView {
        // Create a simple native ad view that matches your news article layout
        val context = layoutInflater.context
        val packageName = context.packageName
        val layoutId = context.resources.getIdentifier("native_ad_news_article", "layout", packageName)
        val adView = layoutInflater.inflate(layoutId, null) as NativeAdView
        
        // Get references to the views
        val headlineView = adView.findViewById<TextView>(context.resources.getIdentifier("ad_headline", "id", packageName))
        val bodyView = adView.findViewById<TextView>(context.resources.getIdentifier("ad_body", "id", packageName))
        val iconView = adView.findViewById<ImageView>(context.resources.getIdentifier("ad_icon", "id", packageName))
        val mediaView = adView.findViewById<com.google.android.gms.ads.nativead.MediaView>(context.resources.getIdentifier("ad_media", "id", packageName))
        val callToActionView = adView.findViewById<Button>(context.resources.getIdentifier("ad_call_to_action", "id", packageName))
        val advertiserView = adView.findViewById<TextView>(context.resources.getIdentifier("ad_advertiser", "id", packageName))
        
        // Populate the views with ad content
        headlineView?.text = nativeAd.headline
        bodyView?.text = nativeAd.body
        callToActionView?.text = nativeAd.callToAction
        advertiserView?.text = nativeAd.advertiser
        
        // Set the icon
        if (nativeAd.icon != null) {
            iconView?.setImageDrawable(nativeAd.icon?.drawable)
            iconView?.visibility = View.VISIBLE
        } else {
            iconView?.visibility = View.GONE
        }
        
        // Set the media content
        mediaView?.setMediaContent(nativeAd.mediaContent)
        
        // Register the views with the native ad
        adView.headlineView = headlineView
        adView.bodyView = bodyView
        adView.iconView = iconView
        adView.mediaView = mediaView
        adView.callToActionView = callToActionView
        adView.advertiserView = advertiserView
        
        // Set the native ad
        adView.setNativeAd(nativeAd)
        
        return adView
    }
}